test/internals/wallet-deployer/migrate_wallet_test.go

```diff
@@ -0,0 +1,267 @@
+	"github.com/tokencard/ethertest"
+)
+
+var _ = Describe("Migrate Wallet", func() {
+
+    var MigratedWallet *bindings.Wallet
+
+	When("no wallets are cached and a controller migrates a Wallet with non-initialized security settings", func() {
+
+		BeforeEach(func() {
+			tx, err := WalletDeployer.MigrateWallet(Controller.TransactOpts(), Owner.Address(), RandomAccount.Address(), false, false, false, EthToWei(2), FinneyToWei(1), []common.Address{common.HexToAddress("0x1"), common.HexToAddress("0x2")})
+			Expect(err).ToNot(HaveOccurred())
+			Backend.Commit()
+			Expect(isSuccessful(tx)).To(BeTrue())
+
+            MigratedWalletAddress, err := WalletDeployer.DeployedWallets(nil, Owner.Address())
+
+            MigratedWallet, err = bindings.NewWallet(MigratedWalletAddress, Backend)
+            Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should NOT set the whitelist flag", func() {
+			initialized, err := MigratedWallet.IsSetWhitelist(nil)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(initialized).To(BeFalse())
+		})
+
+        It("should NOT set the GasTopUpLimit flag", func() {
+			initialized, err := MigratedWallet.GasTopUpLimitSet(nil)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(initialized).To(BeFalse())
+		})
+
+        It("should NOT set the initializedSpendLimit flag", func() {
+			initialized, err := MigratedWallet.SpendLimitSet(nil)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(initialized).To(BeFalse())
+		})
+
+        It("should NOT add the whitelisted addresses to the whitelist", func() {
+            isWhitelisted, err := MigratedWallet.WhitelistMap(nil, common.HexToAddress("0x1"))
+            Expect(err).ToNot(HaveOccurred())
+            Expect(isWhitelisted).To(BeFalse())
+
+            isWhitelisted, err = MigratedWallet.WhitelistMap(nil, common.HexToAddress("0x2"))
+            Expect(err).ToNot(HaveOccurred())
+            Expect(isWhitelisted).To(BeFalse())
+        })
+
+        It("should NOT update the spend limit to 2 ETH", func() {
+            sl, err := MigratedWallet.SpendLimitValue(nil)
+            Expect(err).ToNot(HaveOccurred())
+            Expect(sl.String()).To(Equal(EthToWei(1).String()))
+        })
+
+        It("should NOT update the top up limit to 1 finney", func() {
+            sl, err := MigratedWallet.GasTopUpLimitValue(nil)
+            Expect(err).ToNot(HaveOccurred())
+            Expect(sl.String()).To(Equal(FinneyToWei(500).String()))
+        })
+
+	})
+
+	When("no wallets are cached and a controller migrates a Wallet", func() {
+
+		BeforeEach(func() {
+			tx, err := WalletDeployer.MigrateWallet(Controller.TransactOpts(), Owner.Address(), RandomAccount.Address(), true, true, true, EthToWei(2), FinneyToWei(1), []common.Address{common.HexToAddress("0x1"), common.HexToAddress("0x2")})
+			Expect(err).ToNot(HaveOccurred())
+			Backend.Commit()
+			Expect(isSuccessful(tx)).To(BeTrue())
+		})
+
+		It("should return new Wallet address", func() {
```

Comment:
It's not quite clear what 'it' is here ... maybe `DeployedWallets(ownerAddress) should return then new Wallet address`